File: backend/ruby_foundations/exercises/small_problems_101/easy_7/staggered_caps_2.rb
require 'pry'

def staggered_case(string)
  cap_switch = false
  result = string.chars.each do |letter|
    if cap_switch
      letter.downcase!
    else
      letter.upcase!
    end
    cap_switch = !cap_switch if letter =~ /[a-zA-Z]/
  end
  result.join
end


puts staggered_case('I Love Launch School!') == 'I lOvE lAuNcH sChOoL!'
puts staggered_case('ALL CAPS') == 'AlL cApS'
puts staggered_case('ignore 77 the 444 numbers') == 'IgNoRe 77 ThE 444 nUmBeRs'

File: frontend/javascript/exercises/string_and_text_processing/staggered_caps_2.js
function staggeredCase(string) {
  var count = -1;
  return string.split('').map(function (char) {
    if (char.match(/[a-z]/i)) {
      count++;
    }
    if (count % 2 === 0) {
      return char.toUpperCase();
    } else {
      return char.toLowerCase();
    }
  }).join('');
}

console.log(staggeredCase('I Love Launch School!') === 'I lOvE lAuNcH sChOoL!');     // 'I lOvE lAuNcH sChOoL!'
console.log(staggeredCase('ALL CAPS') === 'AlL cApS');                  // 'AlL cApS'
console.log(staggeredCase('ignore 77 the 444 numbers') === 'IgNoRe 77 ThE 444 nUmBeRs'); // 'IgNoRe 77 ThE 444 nUmBeRs'